### PR TITLE
Hamdi/store

### PR DESCRIPTION
**Pull request by hamdiallam**
1.) `ctx` back in getProof
2.) `store.Block` doesn't need to implement the RLP interface
3.)  constants for the store names

### CODE REVIEW

File: store/query/utxo.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func NewOutputQuerier(outputStore store.OutputStore) sdk.Querier {
 			return data, nil
 		case QueryTxOutput:
 			if len(path) != 2 {
-				return nil, sdk.ErrUnknownRequest("expected txo/<position>")
+				return nil, sdk.ErrUnknownRequest("expected utxo/<position>")
```
Comment (colin-axner) [inline store/query/utxo.go]: What is being requested isn't necessarily unspent

Comment (colin-axner) [reply store/query/utxo.go]: also it should be `output/<position>`

File: cmd/plasmacli/subcmd/eth/prove.go [hunk 2]
```diff
@@ -23,13 +23,15 @@ var proveCmd = &cobra.Command{
 	Args:  cobra.ExactArgs(2),
 	Long:  "Returns proof for transaction inclusion. Use to exit transactions in the smart contract",
 	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := context.NewCLIContext()
```
Comment (colin-axner) [inline cmd/plasmacli/subcmd/eth/prove.go]: when does `WithTrustNode` get set?

File: cmd/plasmacli/subcmd/eth/exit.go [hunk 3]
```diff
@@ -101,7 +102,8 @@ Transaction Exit Usage:
 		var txBytes, proof, confirmSignatures []byte
 		if viper.GetBool(useNodeF) { // query full node
 			var result *tm.ResultTx
-			result, confirmSignatures, err = getProof(position)
+			ctx := context.NewCLIContext()
```
Comment (colin-axner) [inline cmd/plasmacli/subcmd/eth/exit.go]: pretty sure we need to set `WithTrustNode`

**Comment by AdityaSripal** [comment]
What is the reason for putting `ctx` back in? Is it code-hygeine or a correctness issue?

Also need to understand how `trust-node` flag factors in to commands.

**Comment by hamdiallam** [comment]
I put the ctx back in because i think the caller should be responsible for constructing it's ctx object.

--trust-node is the same as used by default in the cosmos sdk. If it's not set, the --chain-id must be set so that the responses can be verified. The trust-node/chain-id flags are only there if the command deals with any interaction with a full node

File: store/query/utxo.go (hunk 1 above)
Comment (hamdiallam) [reply store/query/utxo.go]: ohh gotcha. I thought `txo` was a typo. I'll change this

Comment (hamdiallam) [reply store/query/utxo.go]: what is the difference between QueryTxInput/QueryTxOutput?

output query becomes very verbose. `/output/output/<pos>`

**Pull request merged**